src/__tests__/refs.test.js

```diff
@@ -0,0 +1,45 @@
+import Brahmos, { render } from '..';
+import { forwardRef, createRef, setRef } from '../refs';
+
+describe('Test all ref utils work', () => {
+  it('should forward ref to a Component', () => {
+    class Component extends Brahmos.Component {
+      render () {
+        return null;
+      }
+    }
+
+    const ForwardedComponent = forwardRef((props, ref) => (
+      <Component ref={ref} />
+    ));
+
+    const componentInstance = render(
+      <ForwardedComponent ref={111} />,
+      document.createElement('div')
+    );
+
+    expect(componentInstance.__nodes.ref).toBe(111);
+  });
+
+  it('should return a ref when createRef is called', () => {
+    const receivedRef = createRef();
+    expect(receivedRef).toEqual({
+      current: null,
+    });
+  });
+
+  it('should set instance as current when setRef called with object type ref', () => {
```

Comment:
setRef is inner API we should do integration test of it. Similar to what you did for forwardRefs.

Reply:
👍